modules/ding_provider/connie_search/src/ConnieTingObjectCollection.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * @file
+ * The ConnieTingObjectCollection class.
+ */
+
+namespace Connie\Search;
+
+use Ting\TingObjectCollectionInterface;
+
+class ConnieTingObjectCollection implements TingObjectCollectionInterface {
+
+  /**
+   * @var \Ting\TingObjectInterface[]
+   */
+  protected $objects;
+
+  /**
+   * ConnieObjectCollection constructor.
+   *
+   * @param \Ting\TingObjectInterface[] $objects
+   *   The objects in the collection.
+   */
+  public function __construct($objects) {
+    $this->objects = $objects;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getObjects() {
+    return $this->objects;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPrimaryObject() {
+    // Get the first array-entry (without modifying the array).
+    $objects = array_values($this->objects);
```

Comment:
1. Yes, but the array belong to the object, so it's the only one that have to deal with it.
2. It's a known idiom. Drupal Core uses it all over the place.

> Not anything huge, but I think this code reads better.
Matter of opinion, I think the other way around. :-)